pytorchfi/core.py

```diff
@@ -101,19 +103,27 @@ def _traverse_model_set_hooks(self, model, layer_types):
                 else:
                     for i in layer_types:
                         if isinstance(layer, i):
+                            # neurons
                             handles.append(
                                 layer.register_forward_hook(self._save_output_size)
                             )
-                            shape.append(layer)
+                            output_shape.append(layer)
+
+                            # weights
+                            weights_shape.append(layer.weight.shape)
             # unpack node
             else:
-                subHandles, subBase = self._traverse_model_set_hooks(layer, layer_types)
+                subHandles, subBase, subWeight = self._traverse_model_set_hooks(
```

Comment:
Use `subhandles` or `sub_handles` to keep variable name style consistent.

Reply:
Fixed